static/app/views/releases/utils/useFetchThresholdsListData.tsx

```diff
@@ -11,33 +19,70 @@ export type HookProps = {
 export default function useFetchThresholdsListData({
   selectedEnvs = [],
   selectedProjectIds = [],
-}: HookProps) {
+}: HookProps = {}) {
   const organization = useOrganization();
-
+  const queryClient = useQueryClient();
   const query: ThresholdQuery = {};
+  const isActivatedAlert = organization.features?.includes('activated-alert-rules');
+
+  if (selectedEnvs.length) query.environment = selectedEnvs;
   if (selectedProjectIds.length) {
     query.project = selectedProjectIds;
   } else {
     query.project = [-1];
   }
-  if (selectedEnvs.length) {
-    query.environment = selectedEnvs;
-  }
 
-  return useApiQuery<Threshold[]>(
-    [
-      `/organizations/${organization.slug}/release-thresholds/`,
+  let onSuccess: ((data: unknown[]) => Threshold[]) | undefined = undefined;
+  let queryKey: ApiQueryKey = [
+    `/organizations/${organization.slug}/release-thresholds/`,
+    {
+      query,
+    },
+  ];
+
+  if (isActivatedAlert) {
+    query.monitor_type = 1;
+    queryKey = [
+      `/organizations/${organization.slug}/alert-rules/`,
       {
         query,
       },
-    ],
-    {
-      staleTime: 0,
-      enabled:
-        (organization.features?.includes('releases-v2-internal') ||
-          organization.features?.includes('releases-v2') ||
-          organization.features?.includes('releases-v2-st')) ??
-        false,
-    }
-  );
+    ];
+
+    onSuccess = (data: unknown[]) => {
```

Comment:
i think we're avoiding onSuccess since its gone in react-query v5 https://tanstack.com/query/latest/docs/framework/react/guides/migrating-to-v5#callbacks-on-usequery-and-queryobserver-have-been-removed

Reply:
It looks like a lot of their recommendations would be to chain off of the fetch request -- how would you feel if i added an optional param to `useApiQuery` like: `onFetchSuccess` that is called like: `fetchDataQuery(api).then((data) => onFetchSuccess?.(data))`? (I could do that separately as well since it could be a wonky change for this PR)

I'm hesitant to do the data transformation with `useMemo` in the view, since this data should always be transformed into the `Threshold` type.

Reply:
you can do it in this hook rather than returning the useApiQuery directly

Reply:
like replays does here https://github.com/getsentry/sentry/blob/master/static/app/utils/replays/hooks/useReplayData.tsx#L95-L98

Reply:
ah, smarts! i hadn't thought of deconstructing the response of useApiQuery like that. thanks so much!